router/http/src/main/scala/io/buoyant/router/http/PathIdentifier.scala

```diff
@@ -12,16 +14,17 @@ case class PathIdentifier(
   baseDtab: () => Dtab = () => Dtab.base
 ) extends RoutingFactory.Identifier[http.Request] {
 
-  def apply(req: http.Request): Future[(Dst, http.Request)] =
+  def apply(req: http.Request): Future[RequestIdentification[Request]] =
     req.path.split("/").drop(1) match {
       case path if path.size >= segments =>
         if (consume) {
           req.uri = req.uri.split("/").drop(segments + 1).mkString("/", "/", "")
         }
+        val dst = Dst.Path(prefix ++ Path.Utf8(path.take(segments): _*), baseDtab(), Dtab.local)
+        Future.value(new IdentifiedRequest[Request](dst, req))
+      case _ =>
         Future.value(
-          (Dst.Path(prefix ++ Path.Utf8(path.take(segments): _*), baseDtab(), Dtab.local), req)
+          new UnidentifiedRequest("not enough segments in path")
```

Comment:
If we don't actually do any string formatting here, we can just make this exception static